src/db_window.rs

```diff
@@ -14,15 +14,57 @@ use solana_metrics::{influxdb, submit};
 use solana_sdk::pubkey::Pubkey;
 use std::borrow::Borrow;
 use std::cmp;
+use std::cmp::max;
 use std::net::SocketAddr;
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, RwLock};
 
 pub const MAX_REPAIR_LENGTH: usize = 128;
 
+pub fn generate_repairs(db_ledger: &DbLedger, max_repairs: usize) -> Result<Vec<(u64, u64)>> {
+    // Slot height and blob indexes for blobs we want to repair
+    let mut repairs: Vec<(u64, u64)> = vec![];
+    let mut slots = vec![0];
+    while repairs.len() < max_repairs && !slots.is_empty() {
+        let slot_height = slots.pop().unwrap();
+        let slot = db_ledger.meta(slot_height)?;
+        if slot.is_none() {
+            continue;
+        }
+        let slot = slot.unwrap();
+        slots.extend(slot.next_slots.clone());
+
+        if slot.contains_all_ticks(slot_height, db_ledger) {
```

Comment:
contains_all_ticks() is really "is_complete()"?   meta.blocks is only incremented when a block is completed?